### ISSUE TITLE
Unable to update Danfoss Ally TRV

### ISSUE THREAD

**Issue by johnsprogs** [dresden-elektronik/deconz-ota-plugin#32]
I'm trying to update my Danfoss Ally TRV's from version 1.02 to 1.08. When uploading is ready, progress reports 'done'. But after a minute or so, the upload starts all over again. I'm running DeCONZ 2.13.2. First tried it using the DeCONZ installation on my Domoticz Raspberry Pi. Then installed a fresh DeCONZ on a Windows machine, tried again from there, but with the same result.
Here is my setting:

![afbeelding](https://user-images.githubusercontent.com/75731718/141835523-9d03ff0c-aba7-4b49-821d-3e375fd07ba1.png)

The debug log reports following at the end of an upload:

`
17:47:44:260 Img block req fwVersion:0x00000108, offset: 0x00036F60, maxsize: 63
17:47:44:260 otau send img block rsp offset: 0x00036F60 dataSize 14 0x14B457FFFE76A912
17:47:45:268 otau upgrade end req: status: 0x96, fwVersion:0x00000108, imgType: 0x0100

17:49:40:652 otau query next img req: 0x842e14fffe5e3ece mfCode: 0x1246, img type: 0x0100, sw version: 0x00000102
17:49:40:652 Send query next image response: OTAU_NO_IMAGE_AVAILABLE
17:50:33:017 otau query next img req: 0x842e14fffe5e3ed5 mfCode: 0x1246, img type: 0x0100, sw version: 0x00000102
17:50:33:017 Send query next image response: OTAU_NO_IMAGE_AVAILABLE

17:52:47:576 otau invalide update end request for node 0x14b457fffe76a912

17:52:47:576 otau query next img req: 0x14b457fffe76a912 mfCode: 0x1246, img type: 0x0100, sw version: 0x00000102

17:52:47:576 OTAU: C:/Users/ELJO/otau/1246-0100-01080108.0002_(DF4ECCE1).ota: 225134 bytes
17:52:47:576 OTAU:   ota header (56 bytes)
17:52:47:576 OTAU:   tag 0x0000, length 0x00036F30 (225078 bytes)
17:52:47:576 OTAU:   packed 160968328 bytes
17:53:12:414 Img block req fwVersion:0x00000108, offset: 0x00000000, maxsize: 63
17:53:12:414 otau send img block rsp offset: 0x00000000 dataSize 28 0x14B457FFFE76A912
`
My questions to start with are:

Line 8 states: 'invalid update end request'. Says who? DeCONZ or the TRV?

Line 4 states 'otau query next img req:'. When no uploading is being done, this request comes about every minute. And am I right that it is requesting for version 1.02? But as far as I know, that's already there.

The otau cluster:

![afbeelding](https://user-images.githubusercontent.com/75731718/141837026-6ae0e08d-14b3-41af-af46-6ff61c4a0091.png)

![afbeelding](https://user-images.githubusercontent.com/75731718/141837184-2fc925be-d80c-486c-89ac-e0c74b2b68fa.png)

And shouldn't some of this figures below appear in the second frame (Upgrade and respomse) of the otau-cluster?

![afbeelding](https://user-images.githubusercontent.com/75731718/141837679-360e7d29-456b-4fb8-ba36-fdfb639ff6de.png)

Or do I have to transfer this figures manually and then press execute after a full upload?

If you want more specific info, or want me to try something, just let me know.

Thanks in advance for your effort.

**Comment by ebaauw** [comment]
> Line 8 states: 'invalid update end request'. Says who? DeCONZ or the TRV?

No, it doesn’t.  The OTAU Plugin invalidates the update, after the TRV ended the upgrade on line 2.  Looks like the TRV didn’t like the block it received before.  Did you double check the checksum of the image file?

> Line 4 states 'otau query next img req:'. When no uploading is being done, this request comes about every minute. And am I right that it is requesting for version 1.02? But as far as I know, that's already there.

The TRV reports it’s current firmware in the request for a next version.  It wouldn’t know what next version to request, would it?  Note that this request comes from a different TRV.  I think the plugin sends a no image available since it’s still upgrading the other TRV.  It only handles one device at a time, not to saturate the Zigbee network.
